includes/class-wc-gateway-stripe.php

```diff
@@ -694,7 +712,11 @@ public function process_payment( $order_id, $retry = true, $force_save_source =
 				$new_stripe_customer->create_customer();
 			}
 
-			$prepared_source = $this->prepare_source( get_current_user_id(), $force_save_source );
+			try {
+				$prepared_source = $this->prepare_source_from_intent( get_current_user_id(), $force_save_source );
+			} catch ( WC_Stripe_Exception $e ) {
+				$prepared_source = $this->prepare_source( get_current_user_id(), $force_save_source );
```

Comment:
ToDo: Add some logging here, this "fallback" won't be acceptable in the EU after September.

Reply:
I added `// ToDo: Remove all direct access to a source in favor of intents.` to remind us that this should be done.